UIPS/UIP-0120.md

```diff
@@ -0,0 +1,79 @@
+---
+uip: "013X"
+title: Assertion Rune
+description: "Add a ~: sigcol rune to handle positive assertions cleanly"
+author: ~lagrev-nocfep (@sigilante)
+status: Draft
+type: Standards Track
+category: Hoon
+created: 2024-03-25
+---
+
+## Abstract
+
+Hoon supports several related runes for error handling and debugging:
+
+- `~&` is the classic `printf` side-effect rune.
+- `~|` adds a stack annotation in case of crash.
+- `~?` adds a debugging printf
+- `?>` is a positive assertion (crash if fail).
+- `?<` is a negative assertion (crash if pass).
+
+A common design pattern in Hoon is to have an assertion to succeed, then to wrap
+an intended error mesage with `~|` around `!!`.  By coupling the `?:`, `~|`, and
+`!!` together, we can achieve a more lapidary yet still clear code expression.
```

Comment:
I'd make it clearer here that the big advantage of this rune is that it avoids the footgun of the assertion passing but the message still printing.